### PR TITLE
Suggested updates to KubeSpawner PR about kubernetes_asyncio

### PR DESCRIPTION
**Pull request by consideRatio**
This is a PR to update a PR (https://github.com/jupyterhub/kubespawner/pull/563).

The key change to review is in 96a7358 as it directly influences recently added code. Note that it also has a comment in it explaining the change idea.

I also removed something I think is irrelevant now, but seemed to have been irrelevant even before this PR, that is in 7c55baf. It would be good to have an agreement that it probably is irrelevant and can be safely removed.

### CODE REVIEW

File: kubespawner/clients.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
-"""Shared clients for kubernetes
+"""Configures and instanciates REST API clients of various kinds to
```
Comment (athornton) [inline kubespawner/clients.py:1]: "instantiates"; same two lines down.  Hey, don't blame me, blame the English language.

File: kubespawner/clients.py [hunk 2]
```diff
@@ -54,9 +61,14 @@ def shared_client(ClientType, *args, **kwargs):
...
-    Loads configuration for the Python client we use to communicate with a
-    Kubernetes API server, and optionally tweaks that configuration based
-    on specific settings on the passed caller object.
+    Loads global configuration for the Python client we use to communicate with
+    a Kubernetes API server, and optionally tweaks that configuration based on
+    specific settings on the passed caller object.
+
+    This needs to be called before creating a kubernetes client, so practically
+    before the shared_client function is called. The caller could be KubeSpawner
+    or KubeIngressProxy that both have `k8s_api_ssl_ca_cert` and `k8s_api_host`
```
Comment (athornton) [inline kubespawner/clients.py:70]: Maybe something like "The caller must have both the `k8s_api_ssl_ca_cert` and `k8s_api_host` attributes.  `KubeSpawner` and `KubeIngressProxy` both have these attributes."

File: kubespawner/reflector.py [hunk 3]
```diff
@@ -169,10 +168,12 @@ class ResourceReflector(LoggingConfigurable):
...
         super().__init__(*args, **kwargs)
-        parent = kwargs.get("parent", None)
-        if parent is None:
-            raise RuntimeError("Reflector's parent must be specified!")
-        self.parent = parent
+
+        # Client configuration for kubernetes, as done via the load_config
+        # function, has already taken place in KubeSpawner or KubeIngressProxy
+        # initialization steps.
+        self.api = shared_client(self.api_group_name)
```
Comment (athornton) [inline kubespawner/reflector.py:175]: Oh, OK, I get it now!  Sure, we can get a shared_client and assume that load_config already happened.  I just...somehow didn't understand what you'd intended.

**Review by athornton — APPROVED** [review]
I like it.  A couple of spelling updates, and maybe some phrasing clarification, but on the whole, no objections.

**Comment by consideRatio** [comment]
Thanks for the review @athornton! Usually I'd push a commit to a PR, but I didn't have the rights to do so for your branch in the lsst-sqre repo. For this repo and PR to the specific branch, I also don't have access to press merge - it has to be you.

I'll address your review comments and get back to you in 5 min!

**Review by athornton — APPROVED** [review]
LGTM

**Pull request merged**